shared/test/test_assets.rb

```diff
@@ -154,6 +154,20 @@ def test_assets_quota
     end
   end
 
+  def test_asset_last_modified
+    channel = create_channel(@channels)
+
+    start_time = DateTime.now
+    put @assets, channel, 'test.png', 'not a real image', 'image/png'
+
+    get @assets, channel, 'test.png', '', 'HTTP_IF_MODIFIED_SINCE' => start_time.httpdate
```

Comment:
The If-Modified-Since isn't right here because the server might compare this to the modification time of the resource and decide to return a 304 here, which would break our test.